app/models/post.rb

```diff
@@ -111,6 +111,15 @@ def first_unread_for(user)
     @first_unread ||= reply
   end
 
+  def last_seen_reply_for(user)
+    return @last_seen if @last_seen
+    return unless replies.exists? # unlike first_unread_for we don't care about the post
+    viewed_at = last_read(user) || board.last_read(user)
+    return unless viewed_at
+    reply = replies.where('created_at <= ?', viewed_at).order('id desc').first
+    @last_seen ||= reply
```

Comment:
It doesn't affect anything but I don't think this needs `||=`? (`=` should suffice, since you already checked for `@last_seen` existing.)